crates/re_ui/src/list_item2/scope.rs

```diff
@@ -0,0 +1,137 @@
+use once_cell::sync::Lazy;
+
+// TODO(ab): the state is currently very boring, its main purpose is to support the upcoming two-
+// column ListItemContent.
+#[derive(Debug, Clone)]
+pub struct State {
+    /// X coordinate span to use for hover/selection highlight.
+    ///
+    /// Note: this field is not, strictly speaking, part of the state, as it's overwritten with each
```

Comment:
It's not overwritten though, is it? It is pushed on a stack

Reply:
`State` is stored twice:
1) By itself, on a per scope id basis.
2) As part of the global state stack.

Some `State` fields are "unused" in the context of (1), because `list_item_scope()` always overwrite them after loading `State` for their slot (aka (1)). Of course, then the scope pushes the `State` to the stack, and then these fields are used by the list item contents.

I guess the docstring introduces more confusion than anything, I'll adapt it.